pyccel/codegen/printing/ccode.py

```diff
@@ -1191,7 +1191,8 @@ def _print_Return(self, expr):
         if expr.stmt:
             # get Assign nodes form the CodeBlock object expr.stmt.
             last_assign = expr.stmt.get_attribute_nodes(Assign)
-            vars_in_deallocate_nodes = expr.stmt.get_attribute_nodes(Deallocate, excluded_nodes=(Assign,))
+            deallocate_nodes = expr.stmt.get_attribute_nodes(Deallocate)
```

Comment:
Why did you remove `excluded_nodes=(Assign,)`? We can't assign a `Deallocate` so keeping this can make the code run faster as we don't need to descend into an assign tree